// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.
//
// The following only applies to changes made to this file as part of YugabyteDB development.
//
// Portions Copyright (c) YugabyteDB, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
// in compliance with the License.  You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software distributed under the License
// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
// or implied.  See the License for the specific language governing permissions and limitations
// under the License.
//
syntax = "proto2";

package yb;

// Arbitrary protobuf to test writing a containerized protobuf.
message ProtoContainerTestPB {
  required string name = 1;
  required int32 value = 2;
  optional string note = 3;
  repeated string str_list = 4;
  repeated int32 int_list = 5;
}

message TestStringRequiredPB {
  required string text = 1;
}

message TestStringRepeatedPB {
  repeated string text = 1;
}

message TestStringOptionalPB {
  optional string text = 1;
}

enum TestPBEnum {
  FOO = 1;
  BAR = 2;
  BAZ = 3;
}

message UnsignedIntSetTestPB {
  repeated uint32 set = 1;
}
